### PR TITLE
Release 0.11.4: the SDK follows a wallet whose account is still being created

### PR DESCRIPTION
**Pull request by OscarGauss**
## Summary

Supersedes #55. That branch was cut for `0.11.3`, and `@pollar/core@0.11.3` /
`@pollar/react@0.11.3` are already published — so the five commits that landed
after the cut move here, on top of everything #55 carried. Base is `main` and
the diff is the whole range; the 0.11.3 half is unchanged from #55 and reviewed
there.

The new work is the SDK half of background account provisioning: the platform
now creates an end-user's Stellar account after `POST /auth/login` returns, so
a login hands back a wallet whose on-chain account does not exist yet. Nothing
in the SDK could describe that window — the wallet looked ready, and the first
payment in it came back as an opaque network failure.

### `@pollar/core`

- **`wallet.provisioning`** (`READY | CREATING | FAILED`) on `getWallet()`,
  `getWallets()` and the persisted session. Only the ACCOUNT is described:
  trustlines are added incrementally over an app's life, so a wallet does not
  leave `READY` because a token was enabled yesterday.
- **`onWalletStateChange(cb)`** fires on the transition and replays the current
  value on subscribe, the same contract as `onAuthStateChange`, so a late
  subscriber never waits for a transition that already happened.
- While a wallet is `CREATING` the client polls `GET /wallet/state` — 1s, 2s, 3s
  to a 10s ceiling, at most 12 checks. Linear, not exponential: doubling would
  put the fourth check at 15s, so an account ready at 6s would look unready for
  another nine. Stops on `READY`/`FAILED`, on logout and on `destroy()`. Giving
  up is safe — the next login or resume re-enqueues a creation that never
  landed. `refreshWalletState()` is the manual poke.
- **`isWalletNotReady()` + `WALLET_NOT_READY_CODE`** name the 409 the server
  returns for on-chain operations in that window. Accepts a thrown error or a
  returned tx outcome, since the tx methods report failures as a value.
- **`config.loginTimeoutMs` (45s)** replaces the 10s default on
  `POST /auth/login`. That is the one call where a login does real server-side
  work; under congestion it ran past a minute, so the client aborted while the
  server finished a login nobody was left to receive.
- `schema.d.ts` regenerated against an sdk-api that serves `/wallet/state`. It
  matched the hand-written entry byte for byte, and picked up drift the schema
  already carried (ramp quote expiry/amount, ramp tx `txHash`/`chain`) —
  additive optional fields from earlier sdk-api work.

### `@pollar/react`

- **Fix: a wallet that finishes provisioning now reaches the UI.** Two guards
  swallowed the transition and either alone froze every screen built on it —
  `sessionsEqual` did not compare `provisioning`, and the context memo's
  dependency list did not include it. Both covered by a regression check
  (block 6 of `smoke-react.cjs`), which is what caught the second one.
- **Send** refuses to build a payment while the account is off the ledger and
  says why. Kept apart from the "no transfer path on this network" message:
  collapsing them would tell someone waiting on a brand-new wallet that Stellar
  does not support sending.
- **Receive** warns while the account is being created — the address is valid
  and worth copying, but a payment sent to it now is rejected by the network.
  The one place this window can cost a user a transfer from someone else.
- The wallet button carries the same reason as a banner in its dropdown.
- `walletNotReadyReason(wallet, chain)` is exported so a custom template phrases
  the wait the way the built-in ones do.
- The three template prop types gain an **optional** `notReadyReason`, so a
  custom template written before this keeps compiling untouched.
- **Ramp errors get sentences.** The modal falls back to printing the raw code,
  so a user a cent short read
  `SDK_RAMPS_INSUFFICIENT_BALANCE:stereum:insufficient_usdc (needs 1.88 USDC…)`.
  `SDK_RAMPS_INSUFFICIENT_BALANCE`, `SDK_RAMPS_ONCHAIN_SUBMIT_FAILED` and
  `SDK_RAMPS_ETHERFUSE_ERROR` are now worded like the anchor and Bridge ones
  beside them.

### CI

- A push touching `skills/` leaves a notice and a step summary listing the
  changed files and linking the workflow that publishes them. pollar.xyz mirrors
  that directory to `/.well-known/agent-skills/` on a nightly cron, so an edit
  is live somewhere between minutes and a day later with nothing in this repo
  saying so. A reminder, not a trigger — firing pollar-web's workflow from here
  needs a long-lived cross-repo PAT, a poor trade for saving a few hours.
  `main` only.

## Carried over from #55 (0.11.3)

Session resilience in `@pollar/core` (DPoP-keypair persistence failures, logout
races and cancellation, cross-tab/cross-document ownership), the passkey
ceremony on a consumer-built client, the StrictMode orphan fix, `@pollar/core`
as a **peer dependency only** of `@pollar/react` and the four adapters, the ramp
widget UX and normalized deposit instructions, and five smoke suites. Full
detail in #55.

## State

- Packages are at **0.11.3** in-tree and published on npm. The bump to 0.11.4
  is not in this branch yet.
- The four adapters carry the peer-only change but remain at 0.11.2.
- CHANGELOG has the new work under **Unreleased**.

## Summary by CodeRabbit

- **New Features**
  - Wallet accounts can be created in the background while login completes, with automatic status updates and change notifications.
  - Send, Receive, and wallet controls explain when an account is not ready.
  - Added clearer ramp transaction details, including expiration, amounts, network, transaction hash, and onboarding status.
  - Added configurable login timeout support and wallet readiness detection.

- **Bug Fixes**
  - Fixed React interfaces that could remain stuck while wallet provisioning completed.
  - Improved ramp messaging for provider setup and common transaction failures.
  - Improved recovery from DPoP clock-skew errors without clearing the session.

- **Documentation**
  - Documented wallet provisioning behavior and integration guidance.

### CODE REVIEW

File: packages/core/src/client/client.ts [hunk 1]
```diff
@@ -3962,6 +4126,10 @@ export class PollarClient {
       this._profile = { ...content };
       this._resetResumeBackoff();
       this._setAuthState({ step: 'authenticated', session: this._session, verified: true });
+      // A wallet restored mid-creation resumes its watch here. The server's own
+      // resume re-enqueues a creation that never landed, so by the time the
+      // first check fires there is something to wait for again.
+      this._startProvisioningWatch();
```
Comment (LuaXD) [inline packages/core/src/client/client.ts:4218]: [P1] Start the provisioning flow after an expired-session refresh too

A restored `CREATING` session whose access token is already expired never reaches this line. `_restoreSession()` takes its `expiresAt` branch, awaits `refresh()`, emits `authenticated` with `verified: true`, and returns at line 4022. That path neither calls `_startProvisioningWatch()` nor `/auth/session/resume` (the request documented here as re-enqueuing a stalled creation). Because the session is now verified, the visibility handler will not resume it later either, so `provisioning` can remain `CREATING` and the new UI stays blocked for the lifetime of the client. Please run the same post-validation provisioning step from the expired-token branch, or centralize it so both successful validation paths use it. A regression case with an expired persisted token would catch this.

File: packages/core/src/types.ts [hunk 2]
```diff
@@ -778,6 +811,28 @@ export function isPollarApiError(err: unknown): err is PollarApiError {
...
+ * True when a failure is "the wallet is still being prepared".
+ *
+ * Accepts either a thrown {@link PollarApiError} or a returned transaction
+ * outcome, since the tx methods report failures as a value rather than throwing.
+ * The right response is to wait - {@link PollarClient.onWalletStateChange} fires
+ * when the account lands - not to retry immediately.
+ */
+export function isWalletNotReady(errorOrOutcome: unknown): boolean {
+  if (typeof errorOrOutcome !== 'object' || errorOrOutcome === null) return false;
+  return (errorOrOutcome as { code?: unknown }).code === WALLET_NOT_READY_CODE;
```
Comment (LuaXD) [inline packages/core/src/types.ts:833]: [P2] Preserve the wallet-not-ready code on build outcomes

This helper can only recognize returned outcomes that retain `code`, but `buildTx()` currently reads only `error.details` and `BuildOutcome` has no `code` field. A `/tx/build` 409 with `SDK_WALLET_NOT_READY` therefore becomes `{ status: 'error' }`, and `isWalletNotReady(await client.buildTx(...))` returns false; the external-wallet branch of `buildAndSignAndSubmitTx()` drops the build error the same way. That contradicts the new promise that returned transaction outcomes are accepted. Please preserve `_resolveTxApiError(...).code` in `BuildOutcome` and propagate it through the composed path (plus a 409 regression test).

File: packages/core/src/client/client.ts [hunk 3]
```diff
@@ -1289,6 +1320,137 @@ export class PollarClient {
...
+      // In-memory state is still correct for this process; a storage hiccup must
+      // not cost the app the transition it is waiting for.
+      this._log.error('[PollarClient] Failed to persist wallet provisioning', err);
+    }
+    if (this._destroyed || this._sessionGeneration !== gen) return;
+
+    if (this._authState.step === 'authenticated') {
+      this._setAuthState({ ...this._authState, session: this._session });
+    }
+    for (const cb of this._walletStateListeners) cb(provisioning);
```
Comment (LuaXD) [inline packages/core/src/client/client.ts:1486]: [P2] Isolate wallet-state listeners from each other

If one callback throws here, iteration stops and all later subscribers miss the transition. The throw is caught by `refreshWalletState()`, which returns `null` even though the in-memory state was already changed; the next poll sees the same value, skips emission, returns `READY`, and stops. Thus a single faulty subscriber can permanently prevent every later `onWalletStateChange` subscriber from seeing `READY`. Wrap each callback independently (and log the failure), as `_dispatchStorageDegrade` already does, so the new subscription contract holds for every listener.

File: packages/react/src/components/wallet-button/WalletButton.tsx [hunk 4]
```diff
@@ -43,6 +44,11 @@ export function WalletButton() {
   // only refreshes on the next login/resume).
   const canCreateAccount =
     !created && wallet?.custody === 'external' && wallet.existsOnStellar === false && wallet.fundingMode === 'IMMEDIATE';
+  // A platform-managed wallet whose account is still being created. Surfaced in
+  // the dropdown rather than on the button: the address is real and copyable,
+  // what is missing is the account behind it. `null` for an external wallet,
+  // which the app never provisions (that one gets Create account above).
+  const notReadyReason = walletNotReadyReason(wallet, 'STELLAR');
```
Comment (LuaXD) [inline packages/react/src/components/wallet-button/WalletButton.tsx:51]: [P2] Tie the banner to the chain the wallet button represents

The button displays `primaryAddress` from the app-configured first chain, but this hard-codes the notice to Stellar. In a Solana-first app whose separate Stellar wallet is still `CREATING`, the dropdown beside the Solana address says that "your wallet is still being prepared" even though Solana sends and receives are available. `useChains()` already exposes `primaryChain`; use that here (or explicitly label the banner as Stellar-only) so the status matches the wallet shown by the button.

**Review by LuaXD — COMMENTED** [review]
Reviewed the wallet-provisioning changes at the current head. The clean lockfile build, lint, and full smoke suite pass. I found the following edge cases that are not covered by the current tests.
